test-suite/termstructures.cpp

```diff
@@ -334,6 +335,132 @@ void TermStructureTest::testLinkToNullUnderlying() {
+    rates.push_back(0.0655823213132524);
+    rates.push_back(0.0699455024156877);
+    rates.push_back(0.0799107139233497);
+    rates.push_back(0.0813931951022577);
+    rates.push_back(0.0841615820666691);
+    rates.push_back(0.0501297919004145);
+    rates.push_back(0.0823483583439658);
+    rates.push_back(0.0860720030924466);
+    rates.push_back(0.0922887604375688);
+    rates.push_back(0.10588902278996);
+    rates.push_back(0.117021968693922);
+    rates.push_back(0.109824660896137);
+    rates.push_back(0.109231572878364);
+    rates.push_back(0.119218123236241);
+    rates.push_back(0.128647300167664);
+    rates.push_back(0.0506086995288751);
+
+    boost::shared_ptr<YieldTermStructure> termStructure1 = boost::shared_ptr<YieldTermStructure>(
+        new ForwardCurve(dates, rates, Actual365Fixed(), NullCalendar()));
+
+    // Second curve
+    dates.clear();
+    rates.clear();
+
+    dates.push_back(Date(10, Nov, 2017));
+    dates.push_back(Date(13, Nov, 2017));
+    dates.push_back(Date(11, Dec, 2017));
+    dates.push_back(Date(12, Feb, 2018));
+    dates.push_back(Date(10, May, 2018));
+    dates.push_back(Date(31, Jan, 2022));
+    dates.push_back(Date(7, Dec, 2023));
+    dates.push_back(Date(31, Jan, 2025));
+    dates.push_back(Date(31, Mar, 2028));
+    dates.push_back(Date(7, Dec, 2033));
+    dates.push_back(Date(1, Feb, 2038));
+    dates.push_back(Date(2, Apr, 2046));
+    dates.push_back(Date(2, Jan, 2051));
+    dates.push_back(Date(31, Dec, 2141));
+
+    rates.push_back(0.056656806197189);
+    rates.push_back(0.056656806197189);
+    rates.push_back(0.0419541633454473);
+    rates.push_back(0.0286681050019797);
+    rates.push_back(0.0148840226959593);
+    rates.push_back(0.0246680238374363);
+    rates.push_back(0.0255349067810599);
+    rates.push_back(0.0298907184711927);
+    rates.push_back(0.0263943927922053);
+    rates.push_back(0.0291924526539802);
+    rates.push_back(0.0270049276163556);
+    rates.push_back(0.028775807327614);
+    rates.push_back(0.0293567711641792);
+    rates.push_back(0.010518655099659);
+
+    boost::shared_ptr<YieldTermStructure> termStructure2 = boost::shared_ptr<YieldTermStructure>(
+        new ForwardCurve(dates, rates, Actual365Fixed(), NullCalendar()));
+
+    typedef Real(*binary_f)(Real, Real);
+
+    boost::shared_ptr<YieldTermStructure> compoundCurve = boost::shared_ptr<YieldTermStructure>(
+        new CompositeZeroYieldStructure<binary_f>(Handle<YieldTermStructure>(termStructure1), Handle<YieldTermStructure>(termStructure2), sub));
+
+    // Expected values
+    dates.clear();
+    rates.clear();
+
+    dates.push_back(Date(10, Nov, 2017));
+    dates.push_back(Date(15, Dec, 2017));
+    dates.push_back(Date(15, Jun, 2018));
+    dates.push_back(Date(15, Sep, 2029));
+    dates.push_back(Date(15, Sep, 2038));
+    dates.push_back(Date(15, Mar, 2046));
+    dates.push_back(Date(15, Dec, 2141));
+
+    rates.push_back(0.00892551511527986);
```

Comment:
Are these values what you expect?

Reply:
Yes, the values were independently calculated in Excel.